src/charts/donut.js

```diff
@@ -156,7 +157,7 @@ define(function(require) {
                 if (highlightedSliceId) {
                     initHighlightSlice();
                 }
-                if (isEmpty && emptyDataConfig.showEmptySlice) {
+                if ((isEmpty && emptyDataConfig.showEmptySlice) || shouldShowEmptyState) {
```

Comment:
Actually, I wanted to remove the 'isEmpty' variable altogether.
Also, I think we should get rid of 'emptyDataConfig', keeping only 'emptySliceColor' as a regular accessor.